Source/Plugins/Statistics/StatisticsFilters/FindShapes.cpp

```diff
@@ -338,23 +385,61 @@ void FindShapes::find_moments()
   double konst3 = static_cast<double>((modXRes) * (modYRes) * (modZRes));
   double o3 = 0.0, vol5 = 0.0, omega3 = 0.0;
-  for(size_t i = 1; i < numfeatures; i++)
+  for(size_t featureId = 1; featureId < numfeatures; featureId++)
   {
     // calculating the modified volume for the omega3 value
-    vol5 = m_Volumes[i] * konst3;
-    m_Volumes[i] = m_Volumes[i] * konst2;
-    m_FeatureMoments[i * 6 + 0] = m_FeatureMoments[i * 6 + 0] * konst1;
-    m_FeatureMoments[i * 6 + 1] = m_FeatureMoments[i * 6 + 1] * konst1;
-    m_FeatureMoments[i * 6 + 2] = m_FeatureMoments[i * 6 + 2] * konst1;
-    m_FeatureMoments[i * 6 + 3] = -m_FeatureMoments[i * 6 + 3] * konst1;
-    m_FeatureMoments[i * 6 + 4] = -m_FeatureMoments[i * 6 + 4] * konst1;
-    m_FeatureMoments[i * 6 + 5] = -m_FeatureMoments[i * 6 + 5] * konst1;
-    u200 = static_cast<float>((m_FeatureMoments[i * 6 + 1] + m_FeatureMoments[i * 6 + 2] - m_FeatureMoments[i * 6 + 0]) / 2.0f);
-    u020 = static_cast<float>((m_FeatureMoments[i * 6 + 0] + m_FeatureMoments[i * 6 + 2] - m_FeatureMoments[i * 6 + 1]) / 2.0f);
-    u002 = static_cast<float>((m_FeatureMoments[i * 6 + 0] + m_FeatureMoments[i * 6 + 1] - m_FeatureMoments[i * 6 + 2]) / 2.0f);
-    u110 = static_cast<float>(-m_FeatureMoments[i * 6 + 3]);
-    u011 = static_cast<float>(-m_FeatureMoments[i * 6 + 4]);
-    u101 = static_cast<float>(-m_FeatureMoments[i * 6 + 5]);
+    vol5 = volumes[featureId] * konst3;
+    volumes[featureId] = volumes[featureId] * konst2;
+    featureMoments[featureId * 6 + 0] = featureMoments[featureId * 6 + 0] * konst1;
+    featureMoments[featureId * 6 + 1] = featureMoments[featureId * 6 + 1] * konst1;
+    featureMoments[featureId * 6 + 2] = featureMoments[featureId * 6 + 2] * konst1;
+    featureMoments[featureId * 6 + 3] = -featureMoments[featureId * 6 + 3] * konst1;
+    featureMoments[featureId * 6 + 4] = -featureMoments[featureId * 6 + 4] * konst1;
+    featureMoments[featureId * 6 + 5] = -featureMoments[featureId * 6 + 5] * konst1;
+
+    // Now store the 3x3 Matrix for the Eigen Value/Vectors
+    // clang-format off
+    //featureEigenVals =  Eigen::EigenSolver<Eigen::Matrix3f>::EigenvalueType eigenValues = es.eigenvalues();
+    Eigen::Matrix3f moment;
+    moment(0,0) = featureMoments[featureId * 6 + 0]; /**/ moment(0,1) = featureMoments[featureId * 6 + 3]; /**/moment(0,2) = featureMoments[featureId * 6 + 5];
+    moment(1,0) = featureMoments[featureId * 6 + 3]; /**/ moment(1,1) = featureMoments[featureId * 6 + 1]; /**/moment(1,2) = featureMoments[featureId * 6 + 4];
+    moment(2,0) = featureMoments[featureId * 6 + 5]; /**/ moment(2,1) = featureMoments[featureId * 6 + 4]; /**/moment(2,2) = featureMoments[featureId * 6 + 2];
+    Eigen::EigenSolver<Eigen::Matrix3f> es(moment);
+    Eigen::EigenSolver<Eigen::Matrix3f>::EigenvalueType eigenValues = es.eigenvalues();
+    Eigen::EigenSolver<Eigen::Matrix3f>::EigenvectorsType eigenVectors = es.eigenvectors();
+    // clang-format on
```

Comment:
```suggestion
    Eigen::Matrix3f moment;
    // clang-format off
    moment <<
      featureMoments[featureId * 6 + 0], featureMoments[featureId * 6 + 3], featureMoments[featureId * 6 + 5],
      featureMoments[featureId * 6 + 3], featureMoments[featureId * 6 + 1], featureMoments[featureId * 6 + 4],
      featureMoments[featureId * 6 + 5], featureMoments[featureId * 6 + 4], featureMoments[featureId * 6 + 2];
    // clang-format on
    Eigen::EigenSolver<Eigen::Matrix3f> es(moment);
    Eigen::EigenSolver<Eigen::Matrix3f>::EigenvalueType eigenValues = es.eigenvalues();
    Eigen::EigenSolver<Eigen::Matrix3f>::EigenvectorsType eigenVectors = es.eigenvectors();
```

We can use Eigen's comma initialization here.